modules/github/standardized-repository-settings/main.tf

```diff
@@ -0,0 +1,124 @@
+/**
+  * This module sets up a GitHub repository in a standardized way, to ensure
+  * consistency among things like branch and PR handling, labels and so on.
+  */
+
+terraform {
+  required_version = "0.15.3"
+
+  required_providers {
+    github = {
+      source  = "integrations/github"
+      version = "4.18.0"
+    }
+  }
+}
+
+resource "github_repository" "this" {
+  name        = var.repository_name
+  description = var.repository_description
+
+  allow_auto_merge       = true
+  allow_merge_commit     = true
+  allow_rebase_merge     = false
+  allow_squash_merge     = false
+  delete_branch_on_merge = true
+  has_downloads          = false
+  has_issues             = true
+  has_wiki               = false
+  visibility             = var.repository_visibility
+  vulnerability_alerts   = var.vulnerability_alerts
+}
+
+resource "github_branch_protection" "this" {
+  repository_id          = github_repository.this
+  pattern                = "main"
+  enforce_admins         = true
+  require_signed_commits = true
```

Comment:
This is something that we don't do today. I don't see an issue with it but we need to inform internally.

Reply:
Yes, I kind of snuck that one in there because I like signed commits. I'm ready to take it out though. :)